corehq/apps/reports/standard/ivr.py

```diff
@@ -130,16 +130,12 @@ def rows(self):
         # Had to do this because looking up one document at a time slows things
         # down a lot.
         all_session_ids = xforms_sessions.keys()
-        limit = 500
-        range_max = int(ceil(len(all_session_ids) * 1.0 / limit))
-        for i in range(range_max):
-            lower_bound = i * limit
-            upper_bound = (i + 1) * limit
-            sessions = XFormsSession.view("smsforms/sessions_by_touchforms_id",
-                keys=all_session_ids[lower_bound:upper_bound],
-                include_docs=True).all()
-            for session in sessions:
-                xforms_sessions[session.session_id] = session.submission_id
+        session_submission_map = dict(
+            SQLXFormsSession.objects.filter(session_id__in=all_session_ids).values_list(
+                'session_id', 'submission_id'
+            )
+        )
```

Comment:
this came out quite a bit nicer with SQL.

@gcapalbo - might be worth testing this locally - i didn't have much data